### ISSUE TITLE
Blockers

### ISSUE THREAD

**Issue by mathiastj** [ssypi/gemtd#1]
Hey there

I can see it's been a while since you've been working on this, but I have a question I hoped you could answer:

I have been trying to create gridnav blockers just like you, but as soon as I place them my towers are pushed away, how did you solve this?

**Comment by ssypi** [comment]
Sorry but I don't think I ever had that problem, at least I don't remember anything like that. Just a wild guess, but what's your BaseClass for your towers and how do you create the blocker? I use npc_dota_tower for the BaseClass and [SpawnEntityFromTableSynchronous()](https://github.com/ssypi/gemtd/blob/master/mygem/mygem/scripts/vscripts/Gem.lua#L186) to create the blocker. Oh and do you have true as the third parameter (called findclearspace or something like that) for CreateUnitByName when creating the tower? I remember it not working for buildings and just moving them in middle of the map so it should be set as false.

I guess it's also possible mine are broken as well if they've changed something since like you said it's been a while.

**Comment by mathiastj** [comment]
Thank you for taking the time to answering.
I use npc_dota_creature for my "towers" because I want to equip cosmetics on my models. 
I tryed using npc_dota_tower and it does indeed not move.

Now I'll have to figure out how to get the best of both worlds :)

**Comment by ssypi** [comment]
Would giving them flying movement work to allow pathing over unpathable terrain? Something like spectral dagger or broodmother web uses, not sure if they use flying modifier or what.

**Comment by mathiastj** [comment]
Yeah giving them flying movement worked, but I couldn't get the visual to disappear and it looked a bit silly that my towers were flying around.

But it worked if I just SetAbsOrigin() on the tower right after placing the blocker, so everything's fine! :)
